### PR TITLE
[5.3] AST: Rename getFullName -> getName on ValueDecl & MissingMemberDecl

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
Cherry-pick #31224

### CODE REVIEW

**Comment by AnthonyLatsis** [comment]
apple/llvm-project#1120

@swift-ci please smoke test

**Comment by AnthonyLatsis** [comment]
apple/llvm-project#1120

@swift-ci please test

**Comment by jckarter** [comment]
Thanks @AnthonyLatsis! Let me coordinate with our build managers to get this into the 5.3 branch.

**Review by airspeedswift — APPROVED** [review]
Thanks for picking over to 5.3. Also thanks for adding a link to the original on master. One minor ask, can you summarize what the PR does on follow-up PRs like #31224? That way it isn't necessary to trace back to what it's following up and deduce.

**Comment by AnthonyLatsis** [comment]
> One minor ask, can you summarize what the PR does on follow-up PRs like #31224?

Sure, done! 

@jckarter, could you handle the merging between this and then?

**Comment by jckarter** [comment]
Sorry for the slow reaction, looks like a conflict crept in. Would you be able to clean that up?

**Comment by AnthonyLatsis** [comment]
Oops, give me a sec.

**Comment by AnthonyLatsis** [comment]
apple/llvm-project#1120

@swift-ci please test

**Comment by jckarter** [comment]
Thanks for seeing this through @AnthonyLatsis!
